packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,864 @@
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  final Size? preferredSize;
+
+  /// Constraints for the window. This may not be honored by the platform.
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  final BoxConstraints? constraints;
+}
+
+/// Base class for window controllers.
+///
+/// A window controller must provide a [future] that resolves to a
+/// a [WindowCreationResult] object. This object contains the view
+/// associated with the window, the archetype of the window, the size
+/// of the window, and the state of the window.
+///
+/// The caller may also provide a callback to be called when the window
+/// is destroyed, and a callback to be called when an error is encountered
+/// during the creation of the window.
+///
+/// Each [WindowController] is associated with exactly one root [FlutterView].
+///
+/// When the window is destroyed for any reason (either by the caller or by the
+/// platform), the content of the controller will thereafter be invalid. Callers
+/// may check if this content is invalid via the [isReady] property.
+///
+/// This class implements the [Listenable] interface, so callers can listen
+/// for changes to the window's properties.
+///
+/// {@template flutter.widgets.windowing.experimental}
+/// Do not use this API in production applications or packages published to
+/// pub.dev. Flutter will make breaking changes to this API, even in patch
+/// versions.
+///
+/// This API throws an [UnsupportedError] error unless Flutter’s windowing
+/// feature is enabled by [isWindowingEnabled].
+///
+/// See: https://github.com/flutter/flutter/issues/30701.
+/// {@endtemplate}
+@internal
+abstract class WindowController with ChangeNotifier {
+  @protected
+  /// Sets the view associated with this window.
+  // ignore: use_setters_to_change_properties
+  void setView(FlutterView view) {
+    _view = view;
+  }
+
+  /// The archetype of the window.
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  WindowArchetype get type;
+
+  /// The current size of the window. This may differ from the requested size.
```

Comment:
FYI, Flutter has the convention that the first line should be just a single sentence. The first line is what shows up in IDE's Intellisense popup, multiple lines will make the popup chunky.

```suggestion
  /// The current size of the window.
  ///
  /// This can differ from the requested size.
```